src/pybind/mgr/dashboard_v2/tools.py

```diff
@@ -70,7 +70,51 @@ def _json_error_page(status, message, traceback, version):
                            version=version))
 
 
-class RESTController(object):
+class BaseControllerMeta(type):
+    @property
+    def mgr(cls):
+        """
+        :return: Returns the MgrModule instance of this Ceph dashboard module.
+        """
+        return cls._mgr_module
+
+    @mgr.setter
+    def mgr(cls, value):
+        """
+        :param value: The MgrModule instance of the Ceph dashboard module.
+        """
+        cls._mgr_module = value
+
+    @property
+    def logger(cls):
+        """
+        :return: Returns the logger belonging to the Ceph dashboard module.
+        """
+        return cls.mgr.log
+
+
+class BaseController(six.with_metaclass(BaseControllerMeta, object)):
+    """
+    Base class for all controllers providing API endpoints.
+    """
+    _mgr_module = None
+
+    @property
+    def mgr(self):
```

Comment:
I don't think we need to also re-declare this property here, I think `self.mgr` will be directed to the class method. But you need to confirm this.

Reply:
Found out the instance properties are needed.